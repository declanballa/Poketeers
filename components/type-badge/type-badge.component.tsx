'use client';

import { TypeIcon } from '@components/icon/icon.component';
import styles from '@styles/type-badge.module.scss';
import colors from '@styles/colors.module.scss';

export const TypeBadge = ({ types }: { types: string[]}) => {
  return (
    <>
      {
        types?.map((type: string) =>
          <div
            className={ styles.container }
            key={type}
            style={{ backgroundColor: colors[`${type}Type`]}}>
            <TypeIcon type={ type } />
            <div className={ styles.container__type }>{ type }</div>
          </div> 
        )
      }
    </>
  );
};
