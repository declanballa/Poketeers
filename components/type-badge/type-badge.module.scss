@import '@styles/typography.module.scss';

.container {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 5px;
  gap: 5px;
  border-radius: 3px;  
  float: left;
  margin-right: 1em;
  padding: .5em;

  &__type {
    @include pokemon-type;
  }

}
