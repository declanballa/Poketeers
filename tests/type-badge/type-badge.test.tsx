import 'jsdom-global/register';
import { shallow, ShallowWrapper } from 'enzyme';
import { TypeBadge } from '@components/type-badge/type-badge.component';

describe('when TypeButton component renders', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <TypeBadge types={['grass', 'poison']} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

