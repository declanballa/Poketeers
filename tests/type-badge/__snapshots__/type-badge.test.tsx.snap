// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when TypeButton component renders should match snapshot 1`] = `
<Fragment>
  <div
    className="container"
    key="grass"
    style={
      {
        "backgroundColor": "grassType",
      }
    }
  >
    <TypeIcon
      type="grass"
    />
    <div
      className="container__type"
    >
      grass
    </div>
  </div>
  <div
    className="container"
    key="poison"
    style={
      {
        "backgroundColor": "poisonType",
      }
    }
  >
    <TypeIcon
      type="poison"
    />
    <div
      className="container__type"
    >
      poison
    </div>
  </div>
</Fragment>
`;
